src/app/configuration/configuration.component.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { CouchService } from '../shared/couchdb.service';
+import { ValidatorService } from '../validators/validator.service';
+import { MatStepper } from '@angular/material';
+import { Router } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+
+@Component({
+  templateUrl: './configuration.component.html'
+})
+export class ConfigurationComponent implements OnInit {
+  @ViewChild('stepper') stepper: MatStepper;
+  message = '';
+  select = '';
+  loginForm: FormGroup;
+  configurationFormGroup: FormGroup;
+  contactFormGroup: FormGroup;
+  nations = [];
+
+  constructor(
+    private _formBuilder: FormBuilder,
+    private couchService: CouchService,
+    private validatorService: ValidatorService,
+    private router: Router
+  ) { }
+
+  ngOnInit() {
+    this.loginForm = this._formBuilder.group({
+      username: [ '', Validators.required ],
+      password: [ '', Validators.required ],
+      confirmPassword: [ '', Validators.required ]
+    });
+    this.configurationFormGroup = this._formBuilder.group({
+      planet_type: [ '', Validators.required ],
+      local_domain: [ '', Validators.required ],
+      name: [ '', Validators.required ],
+      parent_domain: [ '', Validators.required ],
+      preferred_lang: [ '', Validators.required ],
+      code: [ '', Validators.required ]
+    });
+    this.contactFormGroup = this._formBuilder.group({
+      firstName: [ '', Validators.required ],
+      lastName: [ '', Validators.required ],
+      middleName: [ '' ],
+      email: [ '', Validators.required ],
+      phoneNumber: [ '', Validators.required ]
+    });
+    this.getNationList();
+  }
+
+  getNationList() {
+    this.couchService.get('nations/_all_docs?include_docs=true')
+      .subscribe((data) => {
+        this.nations = data.rows.map(nations => {
+          return nations.doc;
+        }).filter(nt  => {
+          return nt['_id'].indexOf('_design') !== 0;
+        });
+      }, (error) => this.message = 'There was a problem getting NationList');
+  }
+
+  onChange(selectedValue: string) {
+    this.select = selectedValue;
+    if (this.select === 'community') {
+      this.configurationFormGroup = this._formBuilder.group({
```

Comment:
Since this just changes one field, use [patchValue](https://angular.io/guide/reactive-forms#patchvalue)